src/functions.js

```diff
@@ -12,45 +18,117 @@ function getNumberOfBikes( document ) {
 };
 
 function getAllButtonText( document ) {
-  // get the text inside all tags with class 'btn' and return as an array of strings
-  
+  var buttonText = document.getElementsByClassName('btn');// get the text inside all tags with class 'btn'
+    var buttonList = [];
+
+    for (i=0; i<buttonText.length; i++){
+     
+      buttonList.push(buttonText[i].innerHTML.trim());
+
+    } // and return as an array of strings
+ return buttonList;
+ 
   // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection
   // also you may want to convert the HTMLCollection to an array
+  
 };
 
 function getNavLinksText( document ){
+  var navBar= document.getElementById('navbarSupportedContent');
+  var links= navBar.getElementsByTagName('a');
+  var newArray = [];
+
+  for (i=0; i<links.length; i++){
+    newArray.push(links[i].textContent.trim());
+
+  }
+return newArray;
   // get the text content of all 'A' tags inside element with id 'navbarSupportedContent'
   // reuturn as an array of strings
 }
 
 function addDashesToLinks( document ){
   // surround the text in navigation bar links with '-'
   // no return needed
+  var links = document.querySelectorAll('.nav-link');
+  links.forEach(function(item){
+   
+    item.textContent ='-' + item.textContent + '-';
+  });
+ 
 }
 
 function italicTitles( document ){
-  // convert links in 'Upcoming Events' section to italic using `<i>` tag
-  // no return needed
-}
+
+//console.log(document.all)
```

Comment:
style note: remove commented and unused code, so that there is less 'noise'. Experiment can be kept in different git branches, and old solutions in old commits (that's why commits are good)